lona/html/data_binding/inputs.py

```diff
@@ -104,101 +38,42 @@ def handle_input_event(self, input_event):
         self.value = input_event.data
 
         if self.bubble_up:
-            if input_event.node is self.input_node:
-                input_event.node = self
             return input_event
 
     # properties ##############################################################
     # value
     @property
     def value(self):
-        return self.input_node.attributes.get('value', '')
+        return self.attributes.get('value', '')
 
     @value.setter
     def value(self, new_value):
-        self.input_node.attributes['value'] = new_value
+        self.attributes['value'] = new_value
 
     # disabled
     @property
     def disabled(self):
-        return self.input_nodes.attributes.get('disabled', '')
+        return self.attributes.get('disabled', '')
 
     @disabled.setter
     def disabled(self, new_value):
         with self.lock:
             if new_value:
-                self.input_node.attributes['disabled'] = True
-
-            elif 'disabled' in self.input_node.attributes:
-                self.input_node.attributes.pop('disabled')
+                self.attributes['disabled'] = True
 
-    # input node properties
-    # id
-    @property
-    def id_list(self):
-        return self.input_node.id_list
-
-    @id_list.setter
-    def id_list(self, new_value):
-        self.input_node.id_list = new_value
-
-    # class
-    @property
-    def class_list(self):
-        return self.input_node.class_list
-
-    @class_list.setter
-    def class_list(self, new_value):
-        self.input_node.class_list = new_value
-
-    # style
-    @property
-    def style(self):
-        return self.input_node.style
-
-    @style.setter
-    def style(self, new_value):
-        self.input_node.style = new_value
-
-    # attributes
-    @property
-    def attributes(self):
-        return self.input_node.attributes
-
-    @attributes.setter
-    def attributes(self, new_value):
-        self.input_node.attributes = new_value
-
-    # ignore
-    @property
-    def ignore(self):
-        return self.input_node.ignore
-
-    @ignore.setter
-    def ignore(self, new_value):
-        self.input_node.ignore = new_value
-
-    # events
-    @property
-    def events(self):
-        return self.input_node.events
-
-    @events.setter
-    def events(self, new_value):
-        self.input_node.events = new_value
-
-    # helper
-    @property
-    def has_id(self):
-        return self.input_node.has_id
-
-    @property
-    def has_class(self):
-        return self.input_node.has_class
+            elif 'disabled' in self.attributes:
+                self.attributes.pop('disabled')
 
 
 class TextArea(TextInput):
```

Comment:
I believe `TextArea` is not `TextInput`.
```suggestion
class TextArea(Node):
```
For example now there is a problem: `TextInput` has to have `*args` in constructor just because `TextArea` is not self closing.

Reply:
TextArea has to inherit from TextInput to make the databinding aspect working. Also these two classes do pretty much the same thing. ``SELF_CLOSING_TAG`` for example gets only used by the backend when rendering an ``repr()`` string. The javascript based rendering engine does not use these informations.